code/modules/events/spacevine.dm

```diff
@@ -145,17 +147,50 @@
 	severity = 3
 	quality = NEGATIVE
 
-/datum/spacevine_mutation/aggressive_spread/on_spread(obj/structure/spacevine/holder, turf/target)
-	switch(severity)
-		if(EXPLODE_DEVASTATE)
-			SSexplosions.highturf += target
-		if(EXPLODE_HEAVY)
-			SSexplosions.medturf += target
-		if(EXPLODE_LIGHT)
-			SSexplosions.lowturf += target
+/datum/spacevine_mutation/aggressive_spread/on_spread(obj/structure/spacevine/holder, turf/target, mob/living)
+	for(var/mob/living/M in range(0,get_turf(target))) //Targets and checks all mobs on target turf individually.
+		if(!isvineimmune(M) && M.stat != DEAD) // Don't kill immune creatures. Dead check to prevent log spam when a corpse is trapped between vine eaters.
+			aggrospread_act(holder, M) //aggrospread_act hurts mobs.
 
 /datum/spacevine_mutation/aggressive_spread/on_buckle(obj/structure/spacevine/holder, mob/living/buckled)
-	buckled.ex_act(severity, null, src)
+	aggrospread_act(holder, buckled) //Hurts mobs every time they get buckled. Immunity check is in the entangle proc. The chance for a throw on buckle is weird but kind of works anyway.
+
+/datum/spacevine_mutation/aggressive_spread/aggrospread_act(obj/structure/spacevine/S, mob/living/M) //Proc to severely punish mobs being in the wrong place at the wrong time.
```

Comment:
Use the autodoc format for function descriptions please.
```
/// Does something
/datum/proc/foo()
```

Reply:
I hope I did it right now.